src/Enhavo/Bundle/AppBundle/Mail/MailManager.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+
+namespace Enhavo\Bundle\AppBundle\Mail;
+
+
+use Enhavo\Bundle\MediaBundle\Model\FileInterface;
+use Swift_Mailer;
+use Swift_Message;
+use Symfony\Component\HttpFoundation\File\File;
+use Twig\Environment;
+
+class MailManager
+{
+    /** @var Swift_Mailer */
+    private $mailer;
+
+    /** @var Environment */
+    private $environment;
+
+    /**
+     * @var array
+     */
+    private $config;
+
+    /**
+     * MailManager constructor.
+     * @param Swift_Mailer $mailer
+     * @param Environment $environment
+     * @param array $config
+     */
+    public function __construct(Swift_Mailer $mailer, Environment $environment, array $config)
+    {
+        $this->mailer = $mailer;
+        $this->environment = $environment;
+        $this->config = $config;
+    }
+
+    public function sendMail(string $namespace, $resource, array $attachments = [])
+    {
+        $vars = [
+            'resource' => $resource,
+        ];
+
+        $message = $this->createMessage(
+            $this->config[$namespace]['from'],
+            $this->config[$namespace]['name'],
+            $this->config[$namespace]['to'],
+            $this->config[$namespace]['subject'],
+            $this->config[$namespace]['template'],
+            $vars,
+            $attachments,
+            $this->config[$namespace]['content_type']
+        );
+
+        return $this->mailer->send($message);
+    }
+
+    /**
+     * @param string $from
+     * @param string $senderName
+     * @param string $to
+     * @param string $subject
+     * @param string $template
+     * @param array $context
+     * @param array|FileInterface[]|File[] $attachments
+     * @param string $contentType
+     * @return Swift_Message
+     * @throws \Twig\Error\LoaderError
+     * @throws \Twig\Error\RuntimeError
+     * @throws \Twig\Error\SyntaxError
+     */
+    private function createMessage(string $from, string $senderName, string $to, string $subject, string $template, array $context, array $attachments = [], string $contentType = 'text/plain'): \Swift_Message
+    {
+        $message = new Swift_Message();
+        $message->setSubject($this->renderString($subject, $context))
+            ->setFrom(
+                $this->renderString($from, $context),
+                $this->renderString($senderName, $context)
+            )
+            ->setTo($this->renderString($to, $context))
+        ;
+
+        $template = $this->environment->load($template);
+
+        try {
```

Comment:
Try catch blocks shoudn't be used for controlling structures. I would prefer to check the content type, if it is `multipart/mixed` and then try to render the block separate. Maybe we can add the three different content types as const.